src/base/bittorrent/session.cpp

```diff
@@ -1398,7 +1398,7 @@ void Session::configurePeerClasses()
     // Proactively do the same for 0.0.0.0 and address_v4::any()
     f.add_rule(lt::address_v4::any()
                , lt::address_v4::broadcast()
-               , 1 << lt::session::global_peer_class_id);
+               , 1 << static_cast<std::uint32_t>(lt::session::global_peer_class_id));
```

Comment:
seems possible to use `underlying_index_t` helper in libt 1.2.

Reply:
This way (`static_cast<std::uint32_t>`) is used in libtorrent code itself.
Besides, `underlying_index_t` doesn't exist libtorrent-1.1. Of course, I can use LTUnderlyingType here.

Reply:
or how about adding comment about "we're able to use `underlying_index_t` here" when we dropped libt 1.1?
I don't insist changing anything here, just want to make it easier for maintenance in the future.

Reply:
>I don't insist changing anything here, just want to make it easier for maintenance in the future.

I still preferred to use LTUnderlyingType to change all similar cases in the same way when we drop libt 1.1.